param_linearize.m

```diff
@@ -15,37 +15,43 @@
 n_total_params = prod(n_per_field);
 
 if p_it == 0
-    p = n_total_params;
+    p_out = n_total_params;
     return;
-elseif p_it > n_total_params
+elseif sum(p_it > n_total_params) > 0
```

Comment:
This is fine, but I think the following would be better:

`any(p_it > n_total_params)`